eng/pipelines/templates/stages/archetype-cpp-release.yml

```diff
@@ -8,7 +8,7 @@ stages:
       - stage: Release_${{artifact.safeName}}
         displayName: 'Release ${{artifact.name}}'
         dependsOn: Build
-        condition:  and(succeeded(), ne(variables['Build.Repository.Name'], 'Azure/azure-sdk-for-cpp-pr'))
+        condition:  and(succeeded(), ne(variables['Build.Repository.Name'], 'Azure/azure-sdk-for-cpp'))
```

Comment:
I don't think this condition is correct. Given there is no -pr repo we should just remove the condition.